src/pg/sql/45_observatory_mvt.sql

```diff
@@ -0,0 +1,562 @@
+            (array_agg(distinct geom_geomref_colname))[1] geom_geomref_colnames,
+            (array_agg(distinct geom_tablename||'.'||geom_geomref_colname))[1] geom_geomref_colnames_qualified
+      FROM json_to_recordset(meta)
+      INTO geom_tablenames, geom_colnames, geom_geomref_colnames, geom_geomref_colnames_qualified
+        AS x(id TEXT, numer_id TEXT, numer_aggregate TEXT, numer_colname TEXT, numer_geomref_colname TEXT, numer_tablename TEXT,
+            numer_type TEXT, denom_id TEXT, denom_aggregate TEXT, denom_colname TEXT, denom_geomref_colname TEXT, denom_tablename TEXT,
+            denom_type TEXT, denom_reltype TEXT, geom_id TEXT, geom_colname TEXT, geom_geomref_colname TEXT, geom_tablename TEXT,
+            geom_type TEXT, numer_timespan TEXT, geom_timespan TEXT, normalization TEXT, api_method TEXT, api_args JSON);
+  END IF;
+
+  ---------MC---------
+  IF geography_level = 'us.census.tiger.census_tract' THEN
+    mc_geography_level := 'tract';
+  ELSE
+    mc_geography_level := (string_to_array(geography_level, '.'))[array_length(string_to_array(geography_level, '.'), 1)];
+  END IF;
+
+  mc_table := cdb_observatory.OBS_GetMCTable(mc_schema, mc_geography_level);
+
+  FOREACH mc_category IN ARRAY mc_categories LOOP
+    FOREACH mc_month IN ARRAY mc_months LOOP
+      mc_month_slug := replace(mc_month, '-', '');
+
+      SELECT CASE
+        WHEN mc_category = 'NEP' THEN 'non eating places'
+	      WHEN mc_category = 'EP'  THEN 'eating places'
+	      WHEN mc_category = 'APP' THEN 'apparel'
+	      WHEN mc_category = 'SB'  THEN 'small business'
+	      WHEN mc_category = 'TR'  THEN 'total retail'
+        END INTO mc_category_name;
+
+      SELECT string_agg(column_name||'_'||mc_category||'_'||mc_month_slug, ','),
+            string_agg(mc_table||'_'||mc_category||'_'||mc_month_slug||'.'||column_name||' '||column_name||'_'||mc_category||'_'||mc_month_slug, ','),
+            string_agg(distinct column_name||'_'||mc_category||'_'||mc_month_slug||area_normalization||' '||column_name||'_'||mc_category||'_'||mc_month_slug, ',')
+        INTO numer_colnames_mc_current, numer_colnames_mc_qualified_current, numer_colnames_mc_normalized_current
+        FROM information_schema.columns
+      WHERE table_schema = mc_schema
+        AND table_name = mc_table
+        AND column_name = ANY(mc_measurements);
+
+      IF numer_colnames_mc_current IS NOT NULL THEN
+        numer_colnames_mc := coalesce(numer_colnames_mc, '')||numer_colnames_mc_current||',';
+      END IF;
+      IF numer_colnames_mc_qualified_current IS NOT NULL THEN
+        numer_colnames_mc_qualified := coalesce(numer_colnames_mc_qualified, '')||numer_colnames_mc_qualified_current||',';
+      END IF;
+      IF numer_colnames_mc_normalized_current IS NOT NULL THEN
+        numer_colnames_mc_normalized := coalesce(numer_colnames_mc_normalized, '')||numer_colnames_mc_normalized_current||',';
+      END IF;
+
+      IF mc_table IS NOT NULL THEN
```

Comment:
What if mc_table is null?

Reply:
In this case, the variables below will have the default value and everything will work as if there's only DO data